otherlibs/systhreads/st_stubs.c

```diff
@@ -199,6 +199,7 @@ static inline void caml_thread_save_runtime_state(void)
   curr_thread->backtrace_buffer = Caml_state->backtrace_buffer;
   curr_thread->backtrace_last_exn = Caml_state->backtrace_last_exn;
   curr_thread->memprof_suspended = caml_memprof_suspended;
+  caml_memprof_check_action_pending();
```

Comment:
@jhjourdan it looks like this call is misplaced and should be in `restore_runtime_state`? We assume that you want to call it in this file when `caml_memprof_suspended` is modified, and it is not modified here but in the restoration logic.